torchgeo/datasets/geo.py

```diff
@@ -451,10 +457,14 @@ def __init__(
                         if crs is None:
                             crs = src.crs
 
-                        with WarpedVRT(src, crs=crs) as vrt:
-                            geometries.append(shapely.box(*vrt.bounds))
+                        vrt_options = {'crs': crs}
+                        if self.nodata_value is not None:
+                            vrt_options['nodata'] = self.nodata_value
+                        with WarpedVRT(src, **vrt_options) as vrt:
```

Comment:
Just a heads up, we're possibly getting rid of WarpedVRT in #2945

Reply:
Cool! I'm not too familiar with xarray, but it looks like creating data masks is possible, maybe also it supports writing the masks for caching purposes.

I guess this is what replace WarpedVRT: `src = src.rio.reproject(crs)`